### DISCUSSION TITLE
Growing zone - Allow cutting tree

### DISCUSSION THREAD

**Discussion by Shizen5** [AllenCaine/RimWorld-EnhancementPack#6] [Ideas]
Hi,

This mod add a toggle "Allow harvesting" to growing zone, can it add another toggle "Allow cutting tree" for the new tree lover ideology please?
Or a "Extract tree", instead of cut them, pawn will extract the trees (I'm not sure which choice is better, and this one maybe harder to do).

**Comment by ArcticOwl-Dev** [comment]
I have not played with the tree lover ideology yet. 
When do you have the problem, that you accidently cutting trees? 
The only situation i know is, when you building things and a tree is in the way. 
Maybe i can add a query in the "building" code so they dont automatily cut trees or extract tree instead. 

Is that what you want? 

Alternative i can try to block the cutting of trees completely.

When i have time i look in the code on the weekend.

**Reply by Shizen5** [reply]
Auto cutting trees when building is also an issue, but easier to manage (already have to do this for constructors with very low skill in plants).
I speak about the auto cutting trees in fields, growing zone: there is already a toggle (called "Allow cutting") who disable the cut of tree, but also the cut of plants, grass... So by using this toggle, I have to micro-manage the cut of each plant when my pawns sow, and sometime between the time they harvest and sow back, some grass have time to grow, I have to see it, give order to cut again...

Before, the pawns will also auto chop tree in growing zone ready to be harvest, I think this issue have been already fixed, but I'm not sure.

And yes, you could do a toggle (in right bottom corner of screen) to disable the auto cut of trees (or extract tree instead) for construction and growing zone at the same time. But order to cut/chop tree should still work.

**Comment by ArcticOwl-Dev** [comment]
So I think I got what you want
Please try it out and give feedback.

In the settings under the point AI / behavior you have 2 settings to deactivate all tree cutting or only the automatic cutting in growing zones. 
[TD_EnhancementPack-1.3_Fork_2.3.1pre1.zip](https://github.com/AllenCaine/RimWorld-EnhancementPack/files/7150393/TD_EnhancementPack-1.3_Fork_2.3.1pre1.zip)

**Reply by Shizen5** [reply]
Thank you.

So first, you use 'Tree connection' Meme, instead of the Precepts 'Trees', and 'Cutting trees '. If you want to do this way, the Precepts should be chosen. By example, with fluid Ideo, I started with Collectivist Meme, and with 'Trees: Desired' Precept. Both precepts are not only for 'Tree connection' Meme.
What happen if tree lover guests, or Slave join not tree lover colony?
We should be able to choice to care about guests.
So the toggles should be accessible (screen corner), and to be Save related (by check/uncheck in mod settings, I think the effect apply for all Saves).

"Disable automatic cutting of trees in growing zones" seems to work well, except for the above problem.

About "don't allow cutting trees with tree connection Ideo":
_Good part is it forbid cutting trees because building.
_Bad part is it don't allow the chop wood order to work.
In this state I will never use it.
I'm not sure for 'Cutting trees ' precept, but for 'Trees' precept we can chop some trees without mood debuff. It would be annoying to think to uncheck it each time we want to chop a tree.
Also, Timbershroom and burned trees can be choped without mood debuff.
If you can't disable only tree cutting by buildings, maybe you can made the setting auto toggle off when chop tree order is given?

I saw your answer a bit late, I hope my english is not too bad tonight ^^'
